### ISSUE TITLE
Configuration of the game menu for exporting

### ISSUE THREAD

**Issue by JasonNumberThirteen** [nesbox/TIC-80#1171]
Greetings. I'm so happy I see the next update of software and I have one question.

I would like to ask you about the game menu. Do you consider adding a configuration system for this? I mean, **switching it on/off**, **adjusting menus** (this enabled, that disabled) etc. in case I would like to remove this completely from my export or only some of the options. For example, I would like to keep my content safe and disable the "_Close Game_" option to prevent somebody from looking inside the cartridge. Maybe another pack of variables in the "_config_" cartridge (it's only suggestion)?

Would you like to write what do you think about this?

**Comment by nesbox** [comment]
Hi,
I think, we could describe menu items in the code metadata.

**Comment by JasonNumberThirteen** [comment]
Splendid! Any chance to implement this before 1.0?

**Comment by nesbox** [comment]
I think yes, why not :)

**Comment by RobLoach** [comment]
You could ship one of the other players. SDL, Sokol, the web player, and libretro don't implement the editor.

**Comment by KatPurpy** [comment]
@RobLoach I may be late but I think that SDL player will annoy and confuse people with title "TIC-80 SDL DEMO"
Also TIC-80 players do not implement menus so player wouldn't be able to remap controls as well as reset the game in case it softlocks leading to making them to refresh the page or close and open the program manually

**Comment by borbware** [comment]
I also think that it should be possible to lock some features away to avoid confusion, especially as I'm aiming to ship a commercial game with TIC-80.

My main worries atm are these:

- you can exit the game to the console from the start menu
- you can't reassign controller buttons from the start menu (and you need mouse for the config screen, anyway!)
- you press the start button twice to exit the game to the console, where you suddenly need to use a keyboard in order to return to the game.

**Comment by technomancy** [comment]
By far my favorite thing about TIC is being able to look inside any game I'm playing and see how it works; to tweak things here and there to learn about how other people build things. Of course, TIC is free, and people have the freedom to do with it what they want in their own forks of the codebase, but it would be a shame to take deliberate steps to make the platform less delightful.

**Comment by KatPurpy** [comment]
yeah but considering that players are used to Escape button as pause, exit etc. it makes players' new to TIC80 experience worse

**Comment by borbware** [comment]
> wouldn't mind to still have sources be available but it shouldn't be that easy to accidentally access TIC's console unless there's error

Exactly this. I'm not pushing for a closed experience, but rather for a _smoother_ one: those who just want to play the game can do it without obstructions, and those who want to tinker with the game can still do it - UI just has to make it clear that this is for those who Know What They're Doing.

**Comment by borbware** [comment]
regarding this custom configuration for the menu, I would love to be able to add new menu items in code, like [menuitem](https://pico-8.fandom.com/wiki/Menuitem) in Pico-8.
